flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -194,22 +194,32 @@ public void registerCatalogFunction(
 			UnresolvedIdentifier unresolvedIdentifier,
 			Class<? extends UserDefinedFunction> functionClass,
 			boolean ignoreIfExists) {
-		final ObjectIdentifier identifier = catalogManager.qualifyIdentifier(unresolvedIdentifier);
-		final ObjectIdentifier normalizedIdentifier = FunctionIdentifier.normalizeObjectIdentifier(identifier);
 
 		try {
 			UserDefinedFunctionHelper.validateClass(functionClass);
 		} catch (Throwable t) {
 			throw new ValidationException(
 				String.format(
 					"Could not register catalog function '%s' due to implementation errors.",
-					identifier.asSummaryString()),
+					catalogManager.qualifyIdentifier(unresolvedIdentifier).asSummaryString()),
 				t);
 		}
 
-		final Catalog catalog = catalogManager.getCatalog(normalizedIdentifier.getCatalogName())
-			.orElseThrow(IllegalStateException::new);
-		final ObjectPath path = identifier.toObjectPath();
+		final CatalogFunction catalogFunction = new CatalogFunctionImpl(
```

Comment:
Could you explain the aim of this change? It seems to me there is no difference between the changed logic and the original logic.

Reply:
The initial reason of this change is provide an approach in `FunctionCatalog` to register Python CatalogFunction. But it seems not necessary now because the upper layer accesses the Catalog object directly.